### PR TITLE
Feature/2314 save playlist and regions in screen manager

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to ticket

https://leantime.itkdev.dk/#/tickets/showTicket/2314

#### Description

Related to: https://github.com/os2display/display-api-service/pull/213

- Clean up `screen-manager.jsx`
- Add saving of playlists/groups with screen (as opposed to _after_)
- Change bootstrap column class from `col-md-8` -> `col-md-12` (see image below)
- Update api.generated.ts to match [related pr](https://github.com/os2display/display-api-service/pull/213)
- Fix a small console errer to do with key in `login.jsx`
- [Add @rtk-incubator/rtk-query-codegen-openapi to package.json](https://github.com/os2display/display-admin-client/pull/259/commits/9fc932eebb023c875f593a63a1b350a3b15f0331)
- Sort playlists based on weight in drag/drop component

#### Screenshot of the result

image

### CODE REVIEW

**Review by tuj — APPROVED** [review]

File: src/app.scss [hunk 1]
```diff
@@ -57,13 +57,8 @@ body,
...
   z-index: 1021;
 
-  .spinner-container {
-    display: flex;
-    position: fixed;
-
-    .loading-spinner {
-      margin-right: 1em;
-    }
+  .loading-spinner {
```
Comment (sinejespersen) [inline src/app.scss:60]: Screenshot 2024-10-02 at 20 03 11

File: src/components/user/login.jsx [hunk 2]
```diff
@@ -386,10 +386,10 @@ function Login() {
                     )}
 
                     {usernamePasswordLogins.length > 0 &&
-                      usernamePasswordLogins.map((loginMethod) => (
-                        <div key={loginMethod.provider}>
+                      usernamePasswordLogins.map(({ label, type }) => (
```
Comment (tuj) [inline src/components/user/login.jsx:389]: Why the change from key=loginMethod.provider til loginMethod.type ?

Comment (tuj) [reply src/components/user/login.jsx:389]: This will not result in unique keys

Comment (sinejespersen) [reply src/components/user/login.jsx:389]: Are the types not distinct? Because there were errors in the console stating a unique key was missing.

**Review by tuj — DISMISSED** [review]

**Review by turegjorup — APPROVED** [review]
